iset.mm

```diff
@@ -125615,6 +125689,168 @@ reduced fraction representation (no common factors, denominator
+  Topologies
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $( Token for the class of topologies. $)
+  $c Top $.
+
+  $( Syntax for the class of topologies. $)
+  ctop $a class Top $.
+
+  ${
+    $d x y z $.
+    $( Define the class of topologies.  It is a proper class.  See ~ istopg and
+       ~ istopfin for the corresponding characterizations, using respectively
+       binary intersections like in this definition and nonempty finite
+       intersections.
+
+       The final form of the definition is due to Bourbaki (Def. 1 of
+       [BourbakiTop1] p.  I.1), while the idea of defining a topology in terms
+       of its open sets is due to Aleksandrov.  For the convoluted history of
+       the definitions of these notions, see
+
+       Gregory H. Moore, _The emergence of open sets, closed sets, and limit
+       points in analysis and topology_, Historia Mathematica 35 (2008)
+       220--241.
+
+       (Contributed by NM, 3-Mar-2006.)  (Revised by BJ, 20-Oct-2018.) $)
+    df-top $a |- Top = { x | ( A. y e. ~P x U. y e. x /\
+                             A. y e. x A. z e. x ( y i^i z ) e. x ) } $.
+  $}
+
+  ${
+    $d x y z J $.  $d x A $.
+    $( Express the predicate " ` J ` is a topology."  See ~ istopfin for
+       another characterization using nonempty finite intersections instead of
+       binary intersections.
+
+       Note:  In the literature, a topology is often represented by a
+       calligraphic letter T, which resembles the letter J. This confusion may
+       have led to J being used by some authors (e.g., K. D. Joshi,
+       _Introduction to General Topology_ (1983), p. 114) and it is convenient
+       for us since we later use ` T ` to represent linear transformations
+       (operators).  (Contributed by Stefan Allan, 3-Mar-2006.)  (Revised by
+       Mario Carneiro, 11-Nov-2013.) $)
+    istopg $p |- ( J e. A -> ( J e. Top <-> ( A. x ( x C_ J -> U. x e. J )
+                  /\ A. x e. J A. y e. J ( x i^i y ) e. J ) ) ) $=
+      ( vz wcel ctop cv cuni cpw wral cin wa wss wal wceq pweq eleq2 raleqbi1dv
+      wi anbi12d df-top elab2g df-ral elpw2g imbi1d albidv syl5bb anbi1d bitrd
+      raleqbidv ) DCFZDGFAHZIZDFZADJZKZUMBHLZDFZBDKZADKZMZUMDNZUOTZAOZVAMUNEHZF
+      ZAVFJZKZURVFFZBVFKZAVFKZMVBEDGCVFDPZVIUQVLVAVMVGUOAVHUPVFDQVFDUNRUKVKUTAV
+      FDVJUSBVFDVFDURRSSUAEABUBUCULUQVEVAUQUMUPFZUOTZAOULVEUOAUPUDULVOVDAULVNVC
+      UOUMDCUEUFUGUHUIUJ $.
+
+    $( Express the predicate " ` J ` is a topology" using nonempty finite
+       intersections instead of binary intersections as in ~ istopg .  It is
+       not clear we can prove the converse without adding additional
+       conditions.  (Contributed by NM, 19-Jul-2006.)  (Revised by Jim Kingdon,
+       14-Jan-2023.) $)
+    istopfin $p |- ( J e. Top -> ( A. x ( x C_ J -> U. x e. J )
+         /\ A. x ( ( x C_ J /\ x =/= (/) /\ x e. Fin ) -> |^| x e. J ) ) ) $=
```

Comment:
Depending on the answer to the above comment, we'll know if this is an equivalence or if the second "characterization" is actually too lax for being a definition of "topology".

Reply:
Yeah, like I said the theorems in this pull request are only using the one direction.

Reply:
Of course, because you may have a definition of topology which is too restrictive, making some theorems easier to prove (even if in practice, the difference might occur only for more advanced theorems).